ASF-STM.user.js

```diff
@@ -510,9 +510,9 @@
         throbber.setAttribute("style", "display: none;");
     }
 
-    function updateProgress(index) {
-        let bar = document.getElementById("asf_stm_progress");
-        let progress = 100 * ((index + 1) / bots.Result.length);
+    function updateProgress(index, total, zeroIndex=1) {
+        const bar = document.getElementById("asf_stm_progress");
+        let progress = 100 * index / (total - zeroIndex);
```

Comment:
This is outright wrong. Let's say we have index=5, total=10, zeroIndex=5. With this logic result will be 100%, while it should be 20%. I believe you should remove third parameter and only leave index and total, and rest of calculations should be in the place where you invoke this function.
Also, do you understand that you have division by zero when total=1? You are just lucky that js is shit of a language and don't consider this error, and even allows to set result as element width 🤦‍♂️

Reply:
zeroIndex means if the index starts from 0 or 1, as arrays start from 0 and badge pages start from 1

Reply:
So, zeroIndex=1 means that array starts from 0, and zeroIndex=0 means that array starts from 0? Excuse me, are you high?

Reply:
zeroIndex=0 means that array starts from 1

Reply:
That's some next level bullshit. I won't have such code in my script. Either make it proper or I'll implement it myself, when I'll have some time.